zap/src/main/java/org/zaproxy/zap/extension/search/SearchThread.java

```diff
@@ -279,6 +281,38 @@ private void search() {
                             }
                         }
                     }
+                    if (Type.Note.equals(reqType) && !pcc.allMatchesProcessed()) {
+                        String note = message.getNote();
+                        matcher = pattern.matcher(note);
+
+                        if (inverse && !pcc.allMatchesProcessed()) {
```

Comment:
Drop the right hand side of the expression, already checked earlier.

Reply:
Indeed, good catch.

Done in 002f9f744.